utils/buffer/src/slice.rs

```diff
@@ -110,20 +235,29 @@ impl AsRef<[u8]> for Slice {
 }
 
 impl Drop for Slice {
+    /// Toggles the shared state when the slice is dropped, allowing the memory to be reused.
+    ///
+    /// In debug mode, it also tracks the `mode` of the slice when it is dropped.
     fn drop(&mut self) {
         #[cfg(feature = "debug")]
         self.shared_state.toogle(self.index, self.mode);
+
         #[cfg(not(feature = "debug"))]
         self.shared_state.toogle(self.index);
     }
 }
 
 impl From<Vec<u8>> for Slice {
+    /// Creates a [`Slice`] from a [`Vec<u8>`], taking ownership of the vector.
+    ///
+    /// Initializes the [`Slice`] with the vector's pointer and sets the length to `0`.
```

Comment:
Why did you use public docs comments and not normal ones here?

Reply:
It's public.